WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt

```diff
@@ -155,15 +192,55 @@ class ImprovedMySiteFragment : Fragment(),
                 showSnackbar(messageHolder)
             }
         })
-        viewModel.updateSite((activity as? WPMainActivity)?.selectedSite)
+        viewModel.onMediaUpload.observe(viewLifecycleOwner, {
+            it?.getContentIfNotHandled()?.let { mediaModel ->
+                UploadService.uploadMedia(requireActivity(), mediaModel)
+            }
+        })
         dialogViewModel.onInteraction.observe(viewLifecycleOwner, {
             it?.getContentIfNotHandled()?.let { interaction -> viewModel.onDialogInteraction(interaction) }
         })
+        siteIconUploadViewModel.onUploadedItem.observe(viewLifecycleOwner, {
+            it?.getContentIfNotHandled()?.let { itemUploadedModel ->
+                when (itemUploadedModel) {
+                    is ItemUploadedModel.PostUploaded -> {
+                        uploadUtilsWrapper.onPostUploadedSnackbarHandler(
+                                activity,
+                                requireActivity().findViewById(R.id.coordinator), true,
+                                itemUploadedModel.post, itemUploadedModel.errorMessage, itemUploadedModel.site
+                        )
+                    }
+                    is ItemUploadedModel.MediaUploaded -> {
+                        uploadUtilsWrapper.onMediaUploadedSnackbarHandler(
+                                activity,
+                                requireActivity().findViewById(R.id.coordinator), true,
+                                itemUploadedModel.media, itemUploadedModel.site, itemUploadedModel.errorMessage
+                        )
+                    }
+                }
+            }
+        })
+    }
+
+    private fun startCropActivity(imageUri: UriWrapper) {
+        val context = activity ?: return
+        val options = Options()
+        options.setShowCropGrid(false)
+        options.setStatusBarColor(context.getColorFromAttribute(android.R.attr.statusBarColor))
+        options.setToolbarColor(context.getColorFromAttribute(R.attr.wpColorAppBar))
+        options.setToolbarWidgetColor(context.getColorFromAttribute(attr.colorOnSurface))
+        options.setAllowedGestures(UCropActivity.SCALE, UCropActivity.NONE, UCropActivity.NONE)
+        options.setHideBottomControls(true)
+        UCrop.of(imageUri.uri, Uri.fromFile(File(context.cacheDir, "cropped_for_site_icon.jpg")))
+                .withAspectRatio(1f, 1f)
+                .withOptions(options)
+                .start(requireActivity(), this)
     }
 
     override fun onResume() {
         super.onResume()
-        viewModel.updateSite((activity as? WPMainActivity)?.selectedSite)
+        selectedSiteRepository.updateSite((activity as? WPMainActivity)?.selectedSite)
+        selectedSiteRepository.updateSiteSettingsIfNecessary()
```

Comment:
I'm not sure I'm missing something here, but why are we making calls to the repository directly from the fragment?

Reply:
Good catch, I've fixed it in ac9ab4d

Reply:
hm, when thinking about it, it doesn't really sound like a good idea. I'll move it to the view model 👍